tests/upgrades/test_performance_tuning.py

```diff
@@ -0,0 +1,221 @@
+                                   if data != ''])
+        cmd = 'grep "mongodb::server::storage_engine: \'wiredTiger\'" ' \
+              '/etc/foreman-installer/custom-hiera.yaml'
+        mongodb_type = ssh.command(cmd).return_code
+        if mongodb_type:
+            expected_custom_hiera = set(DEFAULT_CUSTOM_HIERA_DATA + MMPV1_MONGODB)
+        else:
+            expected_custom_hiera = set(DEFAULT_CUSTOM_HIERA_DATA + WIREDTIGER_MONGODB)
+
+        assert expected_custom_hiera == actual_custom_hiera
+
+    @pre_upgrade
+    def test_pre_performance_tuning_apply(self):
+        """In preupgrade scenario we will apply the medium tuning size.
+
+        :id: preupgrade-83404326-20b7-11ea-a370-48f17f1fc2e1
+
+        :steps:
+            1. Run satellite-installer --disable-system-checks.
+
+        :expectedresults: Medium tuning parameter should be applied.
+
+         """
+        cmd = 'grep "mongodb::server::storage_engine: \'wiredTiger\'" ' \
+              '/etc/foreman-installer/custom-hiera.yaml'
+        mongodb_type = ssh.command(cmd).return_code
+        self._custom_hiera_file_creation(mongodb_type, "medium")
+
+        ssh.upload_file('custom-hiera.yaml', '/etc/foreman-installer')
+        command_output = ssh.command('satellite-installer --disable-system-checks',
+                                     connection_timeout=1000).stdout
+        command_output = [data.strip(" ") for data in command_output]
+        assert "Success!" in command_output
+
+    @post_upgrade
+    def test_post_performance_tuning_apply(self):
+        """In postupgrade scenario, we will check the tune parameter state after upgrade
+
+        :id: postupgrade-31e26b08-2157-11ea-9223-001a4a1601d8
+
+        :steps:
+            1. Collect the current tuning state from satellite.yaml file.
+            2. Compare it with the expected value.
+            3. If Expected value get match then this scenario would be passed.
+            1. Run satellite-installer --tuning medium --disable-system-checks.
+
+        :expectedresults: Medium tuning parameter should be applied.
+
+         """
+        cmd = 'grep "mongodb::server::storage_engine: \'wiredTiger\'" ' \
+              '/etc/foreman-installer/custom-hiera.yaml'
+        mongodb_type = ssh.command(cmd).return_code
+        try:
+            self._custom_hiera_file_creation(mongodb_type)
+            cmd = 'grep "tuning: medium" /etc/foreman-installer/scenarios.d/satellite.yaml'
+            tuning_state_after_upgrade = ssh.command(cmd).return_code
+            assert tuning_state_after_upgrade == 0
+            ssh.upload_file('custom-hiera.yaml', '/etc/foreman-installer')
+            command_output = ssh.command('satellite-installer --tuning default '
+                                         '--disable-system-checks',
+                                         connection_timeout=1000).stdout
+            command_output = [data.strip(" ") for data in command_output]
```

Comment:
same here no need to list comprehenssion.

Reply:
Fixed